### PR TITLE
GEOPY-876: validate geoh5 format when using 4.2 save as to 4.1

### PR DESCRIPTION
**Pull request by domfournier**
**GEOPY-876 - Validate geoh5 format when using 4.2 save as to 4.1**

### CODE REVIEW

**Comment by domfournier** [comment]
@sebhmg @jincandescent 
Need to confirm that ga.version == 2.1 is reversed for the new drillhole format using the array "Attributes Jsons", and 2.0 stays with the long "Attributes" json string.

File: geoh5py/shared/concatenation.py [hunk 1]
```diff
@@ -137,13 +136,8 @@ def concat_attr_str(self) -> str:
...
         if self._concat_attr_str is None:
             self._concat_attr_str = "Attributes"
-            if (
-                self.workspace.ga_version is not None
-                and "4." in self.workspace.ga_version
-            ):
-                version = re.findall(r"\d+\.\d+", self.workspace.ga_version)
-                if version and float(version[0]) >= 4.2:
-                    self._concat_attr_str = "Attributes Jsons"
+            if self.workspace.version is not None and self.workspace.version > 2.0:
```
Comment (sebhmg) [inline geoh5py/shared/concatenation.py:139]: **question:** @jincandescent, are there case where the version of Analyst matters to interpret the content of geoh5 ?

**Review by sebhmg — APPROVED** [review]

**Comment by sebhmg** [comment]
🤔**thought:** some of the questions raised during the code review would be solved by introducing a BoundingBox class.

- **allow for various input types:** BoundingBox can be created from a np.ndarray, or 2 tuples or 2 arrays. No need to carry on the Union type in subsequent fonctions. Then it stores and exposes as np.ndarray (or whatever is the simpler / most efficient)
- **checking whether 2D or 3D boxes and combining them:** BoundingBox is always  3D. If proving only x,y, then z is considered -inf, +inf. The code work seamlessly whether initial input  was 2D or 3D
- **checking the coordinates order**: upon creation, BoundingBox orders the points so that the min point has all its component smaller than the max point. No need to test for proper ordering down the road.

File: geoh5py/shared/concatenation.py (hunk 1 above)
Comment (domfournier) [reply geoh5py/shared/concatenation.py:139]: Confirmed with Julien

**Pull request merged**